packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -0,0 +1,678 @@
+   *
+   * @protected
+   */
+  protected deserializeOutputs(): BtcTxOutputTemplate[] {
+    return this._rawTx.outs.map((output, index) => {
+      return new BtcTxOutputTemplate({
+        amountSats: output.value.toString(),
+        address: getOutputAddress(output.script, this._network),
+        locked: index !== this._changeOutputIndex,
+      });
+    });
+  }
+
+  /**
+   * Checks if the transaction signals RBF (Replace-By-Fee).
+   * @returns {boolean} True if the transaction signals RBF, false otherwise
+   * @protected
+   */
+  protected isRBFSignaled(): boolean {
+    if (!this._canRBF) {
+      this._canRBF = this._rawTx.ins.some(
+        (input) => input.sequence < 0xfffffffe,
+      );
+    }
+    return this._canRBF;
+  }
+
+  /**
+   * Determines if CPFP (Child-Pays-For-Parent) can be performed on this transaction.
+   * @returns {boolean} True if CPFP can be performed, false otherwise
+   * @protected
+   */
+  protected canPerformCPFP(): boolean {
+    if (!this._canCPFP) {
+      this._canCPFP = this.outputs.some((output) => output.isMalleable);
+    }
+    return this._canCPFP;
+  }
+
+  /**
+   * Recommends the optimal fee bumping strategy based on the current transaction state.
+   * @returns {FeeBumpStrategy} The recommended fee bumping strategy
+   * @protected
+   */
+  protected recommendStrategy(): FeeBumpStrategy {
+    // TO DO (MRIGESH):
+    // Assuming a tx is non-RBF , but depends on Full-RBF and has lower fees than CPFP then we need to think of mechanism to distinguish a better strategy between the two or maybe warn the user
+
+    if (
+      new BigNumber(this.feeRate).isGreaterThanOrEqualTo(this.targetFeeRate)
+    ) {
+      return FeeBumpStrategy.NONE;
+    }
+
+    const rbfFee = this.minimumRBFFee;
+    const cpfpFee = this.minimumCPFPFee;
+
+    if (
+      this.canRBF &&
+      (!this.canCPFP || new BigNumber(rbfFee).isLessThan(cpfpFee))
+    ) {
+      return FeeBumpStrategy.RBF;
+    } else if (this.canCPFP) {
+      return FeeBumpStrategy.CPFP;
+    }
+
+    return FeeBumpStrategy.NONE;
+  }
+
+  private static validateOptions(
+    options: AnalyzerOptions,
+  ): ValidatedAnalyzerOptions {
+    const validatedOptions: Partial<ValidatedAnalyzerOptions> = {};
+
+    // Raw transaction validation
+    try {
+      const tx = Transaction.fromHex(options.txHex);
+      if (tx.outs.length === 0) {
+        throw new Error("Transaction has no outputs");
+      }
+    } catch (error) {
+      throw new Error(
+        `Invalid transaction: ${error instanceof Error ? error.message : String(error)}`,
+      );
+    }
+    validatedOptions.rawTx = Transaction.fromHex(options.txHex);
+    // Network validation
+    if (!Object.values(Network).includes(options.network)) {
+      throw new Error(`Invalid network: ${options.network}`);
+    }
+    validatedOptions.network = options.network;
+
+    // Target fee rate validation
+    if (
+      typeof options.targetFeeRate !== "number" ||
```

Comment:
This should cover you. If you want better error messages you could check it exists first and throw that it's required. Then you can remove those comments.

Reply:
changes [done](https://github.com/caravan-bitcoin/caravan/pull/114/commits/945d644baf5fcfbfac83271deebe440262046414#diff-b0c4ef170382d75023cd41fe14a8a588936bcc592b31cb86b8abc3178065254fR631)